scripts/shared/lib/deploy_funcs

```diff
@@ -48,18 +48,25 @@ function get_svc_ip() {
 }
 
 function test_connection() {
-    local nginx_address=$1
-    local netshoot_pod
-    netshoot_pod=$(kubectl get pods -l app=netshoot | awk 'FNR == 2 {print $1}')
+    local source_pod=$1
+    local target_address=$2
 
-    echo "Attempting connectivity between clusters - $netshoot_pod cluster2 --> $nginx_address nginx service cluster3"
-    if ! kubectl exec "${netshoot_pod}" -- curl --output /dev/null -m 30 --silent --head --fail "${nginx_address}"; then
+    echo "Attempting connectivity between clusters - $source_pod (cluster2) --> $target_address (service on cluster3)"
+    if ! kubectl exec "${source_pod}" -- curl --output /dev/null -m 30 --silent --head --fail "${target_address}"; then
```

Comment:
I'd take the chance to add --context cluster2 to the exec. So we don't depend on which cluster the caller was... does that make sense?

Reply:
It's already run `with_context` so the caller is responsible for the source cluster. The function itself should be agnostic to this (or it could receive it in params, but for this function it's rather cumbersome)

Reply:
Makes sense, I didn't see the callers :D so I was guessing about that.